sympy/polys/rings.py

```diff
@@ -3041,50 +2054,1014 @@ def subresultants(self, g, x=None):
+                del K[i]
+                poly[tuple(K)] = v
+            else:
+                raise ValueError("Cannot drop generator with non-zero exponent")
+        return poly
+
+    def _drop_to_ground_poly(self, i, ring):
+        poly = ring.zero
+        gen = ring.domain.gens[0]
+        for monom, coeff in self.iterterms():
+            mon = monom[:i] + monom[i + 1:]
+            term = (gen ** monom[i]).mul_ground(coeff)
+            if mon not in poly:
+                poly[mon] = term
+            else:
+                poly[mon] += term
+        return poly
+
+    def to_dense(self):
+        return dmp_from_dict(self, self.ring.ngens-1, self.ring.domain)
+
+    def to_dict(self):
+        return dict(self)
+
+    @property
+    def is_ground(self):
+        return not self or (len(self) == 1 and self.ring.zero_monom in self)
+
+    @property
+    def is_monomial(self):
+        return not self or (len(self) == 1 and self.LC == 1)
+
+    @property
+    def is_zero(self):
+        return not self
+
+    @property
+    def is_squarefree(self):
+        if not self.ring.ngens:
+            return True
+        return self.ring.dmp_sqf_p(self)
+
+    @property
+    def is_irreducible(self):
+        if not self.ring.ngens:
+            return True
+        return self.ring.dmp_irreducible_p(self)
+
+    @property
+    def is_cyclotomic(self):
+        if self.ring.is_univariate:
+            return self.ring.dup_cyclotomic_p(self)
+        else:
+            raise MultivariatePolynomialError("cyclotomic polynomial")
+
+    def _add(self, other):
+        p = self.copy()
+        get = p.get
+        zero = self.ring.domain.zero
+        for k, v in other.iterterms():
+            v = get(k, zero) + v
+            if v:
+                p[k] = v
+            else:
+                del p[k]
+        return p
+
+    def _add_ground(self, cp2):
+        p = self.copy()
+        if not cp2:
+            return p
+        ring = self.ring
+        zm = ring.zero_monom
+        v = self._get_coeff(zm) + cp2
+        if v:
+            p[zm] = v
+        else:
+            del p[zm]
+        return p
+
+    def _sub(self, other):
+        p = self.copy()
+        get = p.get
+        zero = self.ring.domain.zero
+        for k, v in other.iterterms():
+            v = get(k, zero) - v
+            if v:
+                p[k] = v
+            else:
+                del p[k]
+        return p
+
+    def _sub_ground(self, cp2):
+        p = self.copy()
+        if not cp2:
+            return p
+        ring = self.ring
+        zm = ring.zero_monom
+        v = self._get_coeff(zm) - cp2
+        if v:
+            p[zm] = v
+        else:
+            del p[zm]
+        return p
+
+    def _negate(self):
+        p = self.ring.zero
+        for expv in self.itermonoms():
+            p[expv] = -self._get_coeff(expv)
+        return p
+
+    def _rsub_ground(self, n):
+        p = self._negate()
+        return p._add_ground(n)
```

Comment:
I haven't done any timings but these both seem inefficient to me. The existing `__neg__` method would likely be faster than this `_negate` by using a list comprehension and initialising a dict once.

The `rsub_ground` here creates a dict with negate and then another new dict with `add_ground`. The existing code in `__rsub__` seems better.

I think it is better not to change the implementation of things like this unless there is some reason and I don't think there is any reason to change this. In the case of FLINT the implementations here would be different anyway like:
```python
def _negate(self):
    return self._from_rep(-self._rep)

def _rsub_ground(self, n):
    return self._from_rep(n - self._rep)
```